hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/create/ClassicCreatesCall.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.create;
+
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.revertResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.successResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.HtsCall.PricedResult.gasOnly;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.asHeadlongAddress;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.base.TokenType;
+import com.hedera.hapi.node.token.TokenCreateTransactionBody;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.service.contract.impl.exec.scope.VerificationStrategy;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AbstractHtsCall;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AddressIdConverter;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import com.hedera.node.app.service.token.records.CryptoCreateRecordBuilder;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+
+public class ClassicCreatesCall extends AbstractHtsCall {
+
+    @NonNull
+    final TransactionBody syntheticCreate;
+
+    private final AddressIdConverter addressIdConverter;
+    private final VerificationStrategy verificationStrategy;
+    private final org.hyperledger.besu.datatypes.Address spender;
+
+    public ClassicCreatesCall(
+            @NonNull final HederaWorldUpdater.Enhancement enhancement,
+            @NonNull final TransactionBody syntheticCreate,
+            @NonNull final VerificationStrategy verificationStrategy,
+            @NonNull final org.hyperledger.besu.datatypes.Address spender,
```

Comment:
makes sense to me, because from `CreateTranslator` we are passing [attempt.senderAddress()](https://github.com/hashgraph/hedera-services/blob/08591-hts-create-token/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/create/CreateTranslator.java#L155) and it might be confusing (`sender` -> `spender`) in the future. May be it's a good idea to be refactored in another ticket, because this approach is used also for `transfer` and `burn`. What do you think? @tinker-michaelj